Bottom: d1dd5f63a976d72585be5906354cbb110982e60d
Top:    19f87605f79395ff4c00dae5ef6c45f21ee392b3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-05-16 20:47:22 +0100

crosshair.c: Clean up MoveCrosshairAbsolute() to make the logic more readable


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 9a46801..8729237 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -1119,26 +1119,28 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
 bool
 MoveCrosshairAbsolute (Coord X, Coord Y)
 {
-  Coord x, y, z;
-  x = Crosshair.X;
-  y = Crosshair.Y;
+  Coord old_x = Crosshair.X;
+  Coord old_y = Crosshair.Y;
+
   FitCrosshairIntoGrid (X, Y);
-  if (Crosshair.X != x || Crosshair.Y != y)
+
+  if (Crosshair.X != old_x || Crosshair.Y != old_y)
     {
+      Coord new_x = Crosshair.X;
+      Coord new_y = Crosshair.Y;
+
       /* back up to old position to notify the GUI
        * (which might want to erase the old crosshair) */
-      z = Crosshair.X;
-      Crosshair.X = x;
-      x = z;
-      z = Crosshair.Y;
-      Crosshair.Y = y;
+      Crosshair.X = old_x;
+      Crosshair.Y = old_y;
       notify_crosshair_change (false); /* Our caller notifies when it has done */
+
       /* now move forward again */
-      Crosshair.X = x;
-      Crosshair.Y = z;
-      return (true);
+      Crosshair.X = new_x;
+      Crosshair.Y = new_y;
+      return true;
     }
-  return (false);
+  return false;
 }
 
 /* ---------------------------------------------------------------------------
